client/webserver/site/src/js/markets.js

```diff
@@ -1446,14 +1446,43 @@ export default class MarketsPage extends BasePage {
     this.loadTableSide(false)
   }
 
+  /* binOrdersByRateAndEpoch takes a list of sorted orders and returns the
+     same orders grouped into arrays. The orders are grouped by their rate
+     and whether or not they are epoch queue orders. Epoch queue orders
+     will come after non epoch queue orders with the same rate. */
+  binOrdersByRateAndEpoch (orders) {
+    if (!orders || !orders.length) return []
+    const bins = []
+    let currEpochBin = []
+    let currNonEpochBin = []
+    let currRate = orders[0].rate
+    if (orders[0].epoch) currEpochBin.push(orders[0])
+    else currNonEpochBin.push(orders[0])
+    for (let i = 1; i < orders.length; i++) {
+      if (orders[i].rate !== currRate) {
+        bins.push(currNonEpochBin)
+        bins.push(currEpochBin)
+        currEpochBin = []
+        currNonEpochBin = []
+        currRate = orders[i].rate
+      }
+      if (orders[i].epoch) currEpochBin.push(orders[i])
+      else currNonEpochBin.push(orders[i])
+    }
+    bins.push(currNonEpochBin)
+    bins.push(currEpochBin)
+    return bins.filter(bin => bin.length > 0)
+  }
+
   /* loadTables loads the order book side into its table. */
   loadTableSide (sell) {
     const bookSide = sell ? this.book.sells : this.book.buys
     const tbody = sell ? this.page.sellRows : this.page.buyRows
     const cssClass = sell ? 'sellcolor' : 'buycolor'
```

Comment:
`cssClass` is unused outside of `OrderTableRowManager` (via `orderTableRow`). I'd recommend moving this `cssClass` stuff into the `OrderTableRowManager` constructor, and just passing the `sell` boolean as an argument instead.

Reply:
Good call.